WordPress/WordPressApi/WordPressComOAuthClient.swift

```diff
@@ -0,0 +1,189 @@
+import AFNetworking
+
+@objc public enum WordPressComOAuthError: Int {
+    case Unknown
+    case InvalidClient
+    case UnsupportedGrantType
+    case InvalidRequest
+    case NeedsMultifactorCode
+}
+
+/**
+ `WordPressComOAuthClient` encapsulates the pattern of authenticating against WordPress.com OAuth2 service.
+
+ Right now it requires a special client id and secret, so this probably won't work for you
+
+ @see https://developer.wordpress.com/docs/oauth2/
+ */
+
+public class WordPressComOAuthClient: NSObject {
+
+    public static let WordPressComOAuthErrorDomain = "WordPressComOAuthError"
+    public static let WordPressComOAuthKeychainServiceName = "public-api.wordpress.com"
+    public static let WordPressComOAuthBaseUrl = "https://public-api.wordpress.com/oauth2"
+    public static let WordPressComOAuthRedirectUrl = "https://wordpress.com/"
+
+    private let sessionManager: AFHTTPSessionManager
+    private let clientID: String
+    private let secret: String
+
+    /// Creates a WordPresComOAuthClient initialized with the clientID and secret constants defined in the ApiCredentials singleton
+    public class func client() -> WordPressComOAuthClient {
+        let client = WordPressComOAuthClient(clientID:ApiCredentials.client(), secret: ApiCredentials.secret())
+        return client
+    }
+
+    /**
+     Creates a WordPressComOAuthClient using the defined clientID and secret
+
+     - Parameters:
+       - clientID the app oauth clientID
+       - secret the app secret
+     */
+    public init(clientID: String, secret: String) {
+        self.clientID = clientID
+        self.secret = secret
+        let baseURL = NSURL(string:self.dynamicType.WordPressComOAuthBaseUrl)
+        let sessionConfiguration = NSURLSessionConfiguration.ephemeralSessionConfiguration()
+        sessionManager = AFHTTPSessionManager(baseURL:baseURL, sessionConfiguration:sessionConfiguration)
+        sessionManager.responseSerializer = AFJSONResponseSerializer()
+        sessionManager.requestSerializer.setValue("application/json", forHTTPHeaderField:"Accept")
```

Comment:
I'd rather move the session manager initialization to a block in the property declaration. 

From [The Swift Programming Language: Initialization](https://developer.apple.com/library/ios/documentation/Swift/Conceptual/Swift_Programming_Language/Initialization.html):

> If a property always takes the same initial value, provide a default value rather than setting a value within an initializer. The end result is the same, but the default value ties the property’s initialization more closely to its declaration